add direct connect support (#66)
C2Devel/terraform-provider-croccloud: website/docs/d/ec2_transit_gateway_dx_gateway_attachment.html.markdown

---
subcategory: "Transit Gateway"
layout: "aws"
page_title: "aws_ec2_transit_gateway_dx_gateway_attachment"
description: |-

>   Provides information on an EC2 transit gateway's attachment to a Direct Connect gateway.

Reviewer: the attachment of an EC2 transit gateway

Author: Не очень понял, что нужно поменять. Так?
```suggestion
  Provides information about the attachment of an EC2 transit gateway to a Direct Connect gateway.
```

Reviewer: да, так. Можно оставить предлог "on"

Author: Поправил

---

[describe-transit-gateway-attachments]: https://docs.cloud.croc.ru/en/api/ec2/transit_gateways/DescribeTransitGatewayAttachments.html

# Data Source: aws_ec2_transit_gateway_dx_gateway_attachment

> Provides information on an EC2 transit gateway's attachment to a Direct Connect gateway.

Reviewer: the attachment of an EC2 transit gateway 
либо EC2 transit gateway attachment (см. ниже association)

Author: Поправил

## Example Usage

> ### By EC2 transit gateway and Direct Connect gateway identifiers

Reviewer: тут не понял, что хотели сказать с помощью By

Author: Attachment можно получить с помощью следующих аргументов: `transit_gateway_id`, `dx_gateway_id`, `filter`. Все они опциональные. С помощью заголовка пояснили, что мы используем в примере `transit_gateway_id` и `dx_gateway_id`

Reviewer: тут тогда лучше описательно:

 Using EC2 transit gateway and Direct Connect gateway identifiers to get information on the attachment

```terraform
resource "aws_dx_gateway" "example" {

>   name            = "tf-dxg-example"

Reviewer 2: tf-dxgw наверное

Author: Поправил

  amazon_side_asn = "64512"
}

resource "aws_ec2_transit_gateway" "example" {
}

data "aws_ec2_transit_gateway_dx_gateway_attachment" "selected" {
  transit_gateway_id = aws_ec2_transit_gateway.example.id
  dx_gateway_id      = aws_dx_gateway.example.id
}
```

## Argument Reference

The following arguments are supported:

* `transit_gateway_id` - (Optional) The ID of the EC2 transit gateway.
* `dx_gateway_id` - (Optional) The ID of the Direct Connect gateway.

> * `filter` - (Optional) One or more configuration blocks with name-value filters. The structure of this block is [described below](#filter-configuration-block).

Reviewer: The structure of a block
( this означает конкретный (этот) блок)

Author: Поправил

> * `filter` - (Optional) One or more configuration blocks containing name-values filters. The structure of this block is [described below](#filter-configuration-block).

Reviewer: containing => with
name-value

Author: Поправил

### filter Configuration Block

> The following arguments are supported by the `filter` configuration block:

Reviewer: a ... block, их же может быть несколько

Author: Поправил

> * `name` - (Required) The name of the filter field. Valid values can be found in the [EC2 DescribeTransitGatewayAttachments API Reference][describe-transit-gateway-attachments].

Reviewer: после in артикль можно опустить
может просто filter, field выглядит как буквальный перевод

Author: Поправил

> * `values` - (Required) Set of values that are accepted for the given filter field. Results will be selected if any given value matches.

Reviewer: Set of valid values for a given filter field.
matches -- тут по грамматике нужно, с чем совпадает (чему соответствует)

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The ID of the EC2 transit gateway attachment.

> * `tags` - Key-value tags for the EC2 transit gateway attachment.

Reviewer: Key-value кажется здесь лишним
для единообразия можно использовать формулировку как выше

Author: Поправил
